data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/validator/PostAggregateAndProofsV2IntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+
+  @TestTemplate
+  void shouldReturnErros() throws Exception {
+    final SubmitDataError firstSubmitDataError =
+        new SubmitDataError(UInt64.ZERO, "Bad aggregate and proofs");
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(List.of(firstSubmitDataError)));
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef),
+            Collections.emptyMap(),
+            Optional.of(specMilestone.name().toLowerCase(Locale.ROOT)));
+    assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
+
+    final JsonNode resultAsJsonNode = JsonTestUtil.parseAsJsonNode(response.body().string());
+    assertThat(resultAsJsonNode.get("message").asText()).isEqualTo(PARTIAL_PUBLISH_FAILURE_MESSAGE);
+  }
+
+  @TestTemplate
+  void shouldFailWhenMissingConsensusHeader() throws Exception {
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(Collections.emptyList()));
+
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef));
+
+    assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
+
+    final JsonNode resultAsJsonNode = JsonTestUtil.parseAsJsonNode(response.body().string());
+    assertThat(resultAsJsonNode.get("message").asText())
+        .isEqualTo("(Eth-Consensus-Version) header value was unexpected");
+  }
+
+  @TestTemplate
+  void shouldFailWhenBadConsensusHeaderValue() throws Exception {
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(Collections.emptyList()));
+
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef),
+            Collections.emptyMap(),
+            Optional.of("NonExistingMileStone"));
+
+    assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
+
+    final JsonNode resultAsJsonNode = JsonTestUtil.parseAsJsonNode(response.body().string());
+    assertThat(resultAsJsonNode.get("message").asText())
+        .isEqualTo("(Eth-Consensus-Version) header value was unexpected");
```

Comment:
similar here - 'unexpected' is not great. can we have a message like 'the value supplied for Eth-Consensus-Version was invalid: NoExistingMilestone' - then we'd have context...

For me it comes back to a person on discord raising a question with 'Header value was unexpected' - can i help that person?

Reply:
Made it clearer when we're having an unexpected value: `Invalid value for (Eth-Consensus-Version) header: NoExistingMilestone `